plugins/org.eclipse.xtext/src/org/eclipse/xtext/findReferences/AnyTargetURISet.xtend

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2016 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.findReferences
+
+import com.google.common.collect.Iterators
+import com.google.inject.Singleton
+import org.eclipse.emf.common.util.URI
+
+/**
+ * It is used to look up all references for a resource or object scope.
+ * 
+ * @author kosyakov - Initial contribution and API
+ * @since 2.10
+ */
+@Singleton
+package class AnyTargetURISet implements TargetURIs {
+
+	override addAllURIs(Iterable<URI> uris) {
+		// do nothing
```

Comment:
Should we throw `UnsupportedOperationException` in this and the other no-op methods?